### PR TITLE
#15209 Remove agreementText from Java codegen and bindings

### PR DESCRIPTION
**Pull request by hubert-da**
It would probably be easier to make this change in the canton repo first, as there are a bit more changes on the Java bindings side, but that's impossible as Canton depends on `daml-codegen-java` (and this PR is a breaking change for Canton).

### CODE REVIEW

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionGenerator.scala [hunk 1]
```diff
@@ -240,7 +240,6 @@ object TransactionGenerator {
       createdEventBlob,
       interfaceViews.view.collect { case (_, (id, Right(rec))) => (id, rec) }.toMap.asJava,
       interfaceViews.view.collect { case (_, (id, Left(stat))) => (id, stat) }.toMap.asJava,
-      None.toJava, // agreementText, to be removed
```
Comment (stefanobaghino-da) [inline language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionGenerator.scala:243]: Why can't this be removed straight away?

**Review by stefanobaghino-da — COMMENTED** [review]
LGTM but I'll leave to someone from ETX to confirm.

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionGenerator.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionGenerator.scala:243]: We will remove from the proto in a second step. 
I will take care of that.

Comment (hubert-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/TransactionGenerator.scala:243]: This one is actually removed. There are remainings in other places, though.

**Review by mziolekda — APPROVED** [review]

**Pull request merged**